hollow/src/main/java/com/netflix/hollow/api/producer/validation/RecordCountChangeValidator.java

```diff
@@ -0,0 +1,125 @@
+package com.netflix.hollow.api.producer.validation;
+
+import com.netflix.hollow.api.consumer.data.AbstractHollowDataAccessor;
+import com.netflix.hollow.api.producer.HollowProducer;
+import com.netflix.hollow.core.read.engine.HollowReadStateEngine;
+import com.netflix.hollow.core.read.engine.HollowTypeReadState;
+import java.util.logging.Logger;
+
+import static java.util.Objects.requireNonNull;
+
+public class RecordCountChangeValidator implements ValidatorListener {
```

Comment:
naming: since we're observing percent change lets call it `RecordCountPercentChangeValidator`